lunabot_embedded/src/teensy_driver_node.cpp

```diff
@@ -39,27 +87,40 @@ void recv(ros::Publisher &pub) {
     state_msg.dep_curr = state.dep_curr;
     state_msg.exc_curr = state.exc_curr;
     state_msg.act_ang = state.act_ang;
-    state_msg.drive_right_ang = state.drive_right_ang;
-    state_msg.drive_left_ang = state.drive_left_ang;
+    float drive_left_ang_diff =
+        ang_delta(state.drive_left_ang, last_drive_left);
+
+    if (abs(drive_left_ang_diff) >= MAX_DIFF) {
+        state_msg.drive_left_ang = to_rad(last_drive_left);
+    } else {
+        state_msg.drive_left_ang = to_rad(state.drive_left_ang);
+        last_drive_left = state.drive_left_ang;
+    }
+
+    state_msg.drive_right_ang = to_rad(state.drive_right_ang);
     state_msg.lead_screw_ang = state.lead_screw_ang;
-    state_msg.dep_ang = state.dep_ang;
+    state_msg.dep_ang =
+        dep_gear.get_rad_from_raw(state.dep_ang, prev_state.dep_ang);
+    // state_msg.dep_ang = state.dep_ang;
+
+    prev_state = state;
     pub.publish(state_msg);
 }
 
 void effort_cb(const lunabot_msgs::RobotEffort &msg) {
     effort.lead_screw = msg.lead_screw;
     effort.lin_act = msg.lin_act;
-    effort.left_drive = msg.left_drive;
-    effort.right_drive = msg.right_drive;
+    effort.left_drive = -msg.left_drive;
```

Comment:
This inversion can be confusing. Should we have a list of rosparams naming inverted value and using those to multiply by -1 when needed? (or something else to keep better track of this)

Reply:
this was a temporary fix during comp if i remember correctly. Thoughts on having the inversions in preprocessing functions (also similarly abstract out the GearAngle and encoder preprocessing code)

Reply:
Im fine w/ that but the inversions should be placed as a parameter imo. Just easier to read / change.

Reply:
Another thing here is that it should be something we figure out once, then should be the same throughout, so I will be moving this to firmware. see: [this commit](https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43/commits/5f02171344c460dcbfc79db437b1e4a011ac6548)